Feature/user model

Amira578 opened the pull request:
[Added] Delete operation to userController
[Added] Request to handle validation 
[Added] Exception function to handle json response

Amira578 committed:
[Added] Delete Operations to UserController.php
[Added] Route to the Destroy function in routes/api

Amira578 committed:
[Added]Request "UserNewRequest" rules for Validation
[updated] Exception to handle the validations respose

Amira578 committed:
[Created] UserListResource.php
[Created] UserShowResource.php
[Added] getter functions for both phone an created-from attributes
[Updated] UserController.php to use the new Resources functions

Amira578 committed:
[Created] UsersTest.php
[Added] Test cases for CRUD operations and validations

Amira578 committed:
[Created] Project Resources
[Created] Project Requests

Amira578 committed:
[Added] Delete Operation to ProjectController.php

Amira578 committed:
[Updated] Project Routes

aelnemr commented on app/Http/Controllers/API/V1/ProjectController.php:
```diff
@@ -8,17 +8,21 @@
         return $project;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Request Form Validations?

aelnemr commented on app/Http/Controllers/API/V1/ProjectController.php:
```diff
@@ -8,17 +8,21 @@
         return $project;
     }
-    public function update( Request $request, Project $project){
+    public function update( Request $request, Project $project)
```
Request Form Validations?

aelnemr commented on app/Http/Controllers/API/V1/TagController.php:
```diff
@@ -7,17 +7,21 @@
         return $tag;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Request Form Validations?

Amira578 replied:
still didn't do any resources or requests for task and tag models

aelnemr commented on app/Http/Controllers/API/V1/TagController.php:
```diff
@@ -7,17 +7,21 @@
         return $tag;
     }
-    public function update( Request $request, Tag $tag){
+    public function update( Request $request, Tag $tag)
```
Request Form Validations?

aelnemr commented on app/Http/Controllers/API/V1/TaskController.php:
```diff
@@ -8,17 +8,21 @@
         return $task;
     }
-    public function store( Request $request){
+    public function store( Request $request)
```
Request Form Validations?

aelnemr commented on app/Http/Controllers/API/V1/TaskController.php:
```diff
@@ -8,17 +8,21 @@
         return $task;
     }
-    public function update( Request $request, Task $task){
+    public function update( Request $request, Task $task)
```
Request Form Validations?

aelnemr commented on app/Http/Controllers/API/V1/TaskController.php:
```diff
@@ -8,17 +8,21 @@
-    public function index(){
+    public function index()
+    {
         return Task::all();
```
Response resource?

aelnemr commented on app/Http/Controllers/API/V1/ProjectController.php:
```diff
@@ -8,17 +8,21 @@
-    public function index(){
+    public function index()
+    {
         return Project::all();
```
Response resource?

Amira578 committed:
[Added] ProjectFactory.php
[Added] project test cases

aelnemr reviewed (changes requested):
Please add Request Form Validations for all `store`, `update` functions and response resource for all returned objects and lists

Amira578 committed:
[updated] ProjctController

Amira578 committed:
[Added] Tag Resources
[Added] Tag Requests
[Added] Task Resources
[Added] task Requests
[Updated] Route api file
[Added] delete operation for both tag and task controllers

Amira578 committed:
[..]
